src/Block/AdminSearchBlockService.php

```diff
@@ -68,6 +68,10 @@ public function execute(BlockContextInterface $blockContext, Response $response
             $blockContext->getSetting('per_page')
         );
 
+        if (false === $pager) {
+            return null === $response ? new Response() : $response;
```

Comment:
Should the status code be set to 204? Or maybe 205 (should some HTML block be emptied)?

Reply:
From what I can see, there's an empty response going in. And the status code of the returned response isn't used. Only the caching stuff (TTL, age, date) is used.

Reply:
It's not about being used, it's about expressing the fact that it is normal to get an empty response. People might wonder if this is normal or a bug.

Reply:
Oh, wait. The status code is checked to see if the block is cacheable against codes 200, 203, 300, 301, 302, 404, 410.
Then using 204 or 205 will make the block uncacheable.

Reply:
Should it really be cached though? 204 and 205 should be considered cacheable by default, see https://httpstatuses.com/204 and https://httpstatuses.com/205